### PR TITLE
#889 program localization

### LINKED ISSUE #889: localisation for program based stocktakes/requisitions model (ByProgramModel)
Build Number: 2.2.0-RC

Description: IC customisations. ByProgramModal.js needs to use localisation

Reproducible: Yes

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #889 

- Adds localization to newly added features - `byprogramModal`, `StocktakeEditPage` and `StocktakeBatchModal`

Found less in 4D xliff than expected.

Strings required to be translated:

```
fr: {
    select_a_program: 'Select a program',
    select_a_supplier: 'Select a supplier',
    select_an_order_type: 'Select an order type',
    select_a_period: 'Select a period',
    no_programs: 'No programs available',
    no_suppliers: 'No suppliers available',
    no_order_types: 'No order types available',
    no_periods: 'No periods available',
    program: 'programme',
    supplier: 'supplier',
    order_type: 'order type',
    period: 'period',
    max_mos: 'Max MOS',
    threshold_mos: 'Threshold MOS',
    max_orders_per_period: 'Max orders per period',
    requisitions: 'Requisition(s)',
    details: 'Details',
    general: 'General',
    order: 'order',
    not_applicable: 'Not applicable',
  },

   select_a_reason: 'Select a reason',

    by_batch: 'By batch',

    reason: 'REASON',
    batches: 'BATCHES',
    batch_name: 'BATCH NAME',
    expiry: 'EXPIRY',
```

### CODE REVIEW

File: src/localization/modalStrings.js [hunk 1]
```diff
@@ -7,6 +7,10 @@ import LocalizedStrings from 'react-native-localization';
 
 /* eslint-disable max-len */
 
+/**
+ * TODO: select_a_reason
+ */
```
Comment (Chris-Petty) [inline src/localization/modalStrings.js]: We usually add this to each language object as a comment, see elsewhere in this file

File: src/localization/pageInfoStrings.js [hunk 2]
```diff
@@ -3,6 +3,10 @@
  * Sustainable Solutions (NZ) Ltd. 2019
  */
 
+/**
+ * TODO: by_batch
+ */
```
Comment (Chris-Petty) [inline src/localization/pageInfoStrings.js]: Also this

File: src/localization/tableStrings.js [hunk 3]
```diff
@@ -3,6 +3,10 @@
  * Sustainable Solutions (NZ) Ltd. 2019
  */
 
+/**
+ * TODO: reason, batches, batch_name, expiry
+ */
```
Comment (Chris-Petty) [inline src/localization/tableStrings.js]: also this

**Comment by alainsussol** [comment]
@Chris-Petty 
Here are the translations:
[Mobile french translation.xlsx](https://github.com/openmsupply/mobile/files/3136249/Mobile.french.translation.xlsx)

**Comment by Chris-Petty** [comment]
@alainsussol thanks!

**Pull request merged**